### ISSUE TITLE
nvidia-docker installation GPG error

### ISSUE THREAD

**Issue by enricoschroeder** [NVIDIA/nvidia-docker#1658]
Following the installation guide for nvidia-docker (https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#setting-up-nvidia-container-toolkit), I am getting a GPG error when using apt get (see below). This results in me not being able to install nvidia-docker.

There are quite a few issues here already that are related to GPG keys, however they all seem to occur for a different repo (cuda). Thus, the proposed workarounds do not apply here. Please advise on how to install nvidia-docker.

I am using Ubuntu 20.04 LTS.
```
W: GPG error: https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/amd64  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY DDCAE044F796ECB0
E: The repository 'https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/amd64  InRelease' is not signed.
```

**Comment by enricoschroeder** [comment]
Nevermind, it was a strange permissions error on the imported keyring. After following the installation guide, the nvidia keyring had permissions 640 (no anonymous read). Don't know why this was not enough, since apt is run as root, but anyways... I have changed permissions to 644 `sudo chmod 644 /usr/share/keyring/nvidia-container-toolkit-keyring.gpg` and now everything works.

**Comment by ronald-das** [comment]
@enricoschroeder I used the cuda keyring package (https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/) and than tried `sudo apt-get update` , but am getting still getting the same issue `W: GPG error: https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/amd64  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY DDCAE044F796ECB0`, how did you solve this?

**Comment by chm123** [comment]
> @enricoschroeder I used the cuda keyring package (https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/) and than tried `sudo apt-get update` , but am getting still getting the same issue `W: GPG error: https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/amd64 InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY DDCAE044F796ECB0`, how did you solve this?

I also got a similar problem. 
https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#ubuntu
I followed 3.10.3.2 and it worked. I guess you are missing [signed-by=/usr/share/keyrings/cuda-archive-keyring.gpg] as mentioned in 3.10.3.2.b.
